GitUI/CommandsDialogs/CommitDialog/CommitTemplateItem.cs

```diff
@@ -110,7 +113,28 @@ private static CommitTemplateItem[] DeserializeCommitTemplates(string serialized
 
             return commitTemplateItem;
         }
-    
+
+        private static readonly Dictionary<string, Func<string>> RegistratedTemplates = new Dictionary<string, Func<string>>();
+        [NotNull]
+        public static CommitTemplateItem[] LoadFromRegistrated()
+        {
+            return RegistratedTemplates.Select(item => new CommitTemplateItem(item.Key, item.Value())).ToArray();
+        }
+
+        public static bool TryRegistrate(string key, Func<string> item)
+        {
+            if (RegistratedTemplates.ContainsKey(key))
+                return false;
+            RegistratedTemplates.Add(key, item);
+            return true;
+        }
+
+        public static void UnRegistrate(string key)
+        {
+            if (RegistratedTemplates.ContainsKey(key))
+                RegistratedTemplates.Remove(key);
+        }
+
```

Comment:
1. The idea of having a static dictionary seems to me not very good....
2. Concerning the creation in the plugi-n an object of type `CommitTemplateManager`: in which assembly should it be in this case? :)

As I understand the architecture of this application:
- `GitUI` assembly should not be connected to plugins.
- The `IGitUICommands` interface in the `GitUIPluginInterfaces` assembly
- Assembly `GitUIPluginInterfaces` is available for plug-ins
- You can get to the context of `IGitUICommands` via `GitUIBaseEventArgs`

So, I think, it can be done without the interface for the first time.